clang/test/SemaSYCL/esimd-special-class.cpp

```diff
@@ -0,0 +1,28 @@
+// RUN: %clang_cc1 -fsycl-is-device -verify \
+// RUN: -internal-isystem %S/Inputs -triple spir64-unknown-unknown \
+// RUN: -fsyntax-only -disable-llvm-passes -emit-llvm %s -o - %s
+
+// The test is to ensure that the use of sycl_explicit_simd attribute doesn't
+// crash when used with sampler or stream. Currently samplers/stream are not
+// supported in esimd.
+
+#include "sycl.hpp"
+using namespace cl::sycl;
+void test() {
+
+  queue q;
+
+  q.submit([&](handler &h) {
+    cl::sycl::sampler Smplr;
+    cl::sycl::stream Stream(1024, 128, h);
+    // expected-note@+1{{in instantiation of function template specialization}}
+    h.single_task<class esimd_kernel>(
```

Comment:
Nit about coding standard - I'm actually not sure what naming standard applies here but there is inconsistency between `esimd_kernel ` and `StreamTester ` below (capitalized). `SamplerTester `is also a better name IMO